client/state/user-settings/actions.js

```diff
@@ -105,6 +105,27 @@ export const removeUnsavedUserSetting = ( settingName ) => ( {
 	settingName,
 } );
 
+/**
+ * Checks if an incoming change to settings.language is a change to the existing settings
+ * Currently the assumption is that if a settings.locale_variant slug exists, then that is the current language
+ *
+ * @param  {string}  languageSettingValue the newly-set language slug string.
+ * @param  {object}  settings user settings object.
+ * @returns {boolean} if the language setting has been changed.
+ */
+function hasLanguageChanged( languageSettingValue, settings = {} ) {
+	if ( ! languageSettingValue ) {
+		return false;
+	}
+	// if there is a saved variant we know that the user is changing back to the root language === setting hasn't changed
+	// but if settings.locale_variant is not empty then we assume the user is trying to switch back to the root
+	return (
```

Comment:
Should we also consider the empathy mode setting as we do here: https://github.com/Automattic/wp-calypso/pull/49760/files#diff-79489c122f24dcb8752581772ddb5070e083a14f1a77ebbfd4e04c6b2f8c64f8R206

Reply:
That should already be covered. Only the `language` attribute is special because of its relation to `locale_variant`. While testing this I noticed that `i18n_empathy_mode` and `use_fallback_for_incomplete_languages` was missing from `INTERFACE_FIELDS` so I added them in https://github.com/Automattic/wp-calypso/pull/49760/commits/b5cf6150ae2752ae0d6f39d832d13001f90ce9f3